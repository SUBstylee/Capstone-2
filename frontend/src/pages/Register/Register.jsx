import { useState,useEffect } from "react";
import { Link,useNavigate,useLocation } from "react-router-dom";
import {Form,Button,Row,Col} from 'react-bootstrap';
import { useDispatch,useSelector } from "react-redux";
import Message from "../../components/Message/Message";
import Loader from '../../components/Loader/Loader';
import CustomInput from "../../components/CustomInput/CustomInput";
import {register} from '../../actions/userActions';
import MetaWrapper from "../../components/MetaWrapper/MetaWrapper";

import FormContainer from "../../components/FormContainer/FormContainer";
import PasswordGenerator from "../../components/PasswordGenerator/PasswordGenerator";

const Register = () => {
    const [name,setName]=useState('')
    const [email,setEmail]=useState('');
    const [password,setPassword]=useState('');
    const [confirmPassword,setConfirmPassword]=useState('');
    const [message,setMessage]=useState(null);

    const navigate = useNavigate();
    const location = useLocation();
    const dispatch=useDispatch();
    const userRegister=useSelector((state)=>state.userRegister);
    const {loading,error,userInfo}=userRegister;
    const redirect = location.search ? location.search.split('=')[1] : '/';

    useEffect(()=>{
        if(userInfo){
            navigate(redirect);
        };
    },[navigate,userInfo,redirect]);
    
    const submitHandler=(e)=>{
        e.preventDefault();
        if(password!==confirmPassword){
            setMessage('Passwords do not match!')
        }else{
            dispatch(register(name,email,password))
        };
        
    };

  return (
    <div className="Login">
        <Row>
            <Col lg={6}>
        <FormContainer>
            <h1>Sign Up</h1>
            {message&&<Message variant='danger'>{message}</Message>}
            {error&&<Message variant='danger'>{error}</Message>}
            {loading&&<Loader/>}
            <MetaWrapper title='TAA-Register'/>
            <Form onSubmit={submitHandler}>   
                <CustomInput 
                    id='name' 
                    type='text' 
                    autoComplete="current-name" 
                    name='name' 
                    labelText='Full Name' 
                    value={name} 
                    required 
                    onChange={(e)=>setName(e.target.value)}/>
                <CustomInput 
                    id='email' 
                    type='email' 
                    autoComplete="current-email" 
                    name='email' 
                    labelText='Email Address' 
                    value={email} 
                    required 
                    onChange={(e)=>setEmail(e.target.value)}/>      
                <CustomInput 
                    id='password' 
                    type='password' 
                    autoComplete="new-password" 
                    name='password'
                    labelText='Password'
                    value={password} 
                    required 
                    onChange={(e)=>setPassword(e.target.value)}/>   
                <CustomInput
                    id='confirmPassword'
                    type='password'
                    autoComplete="new-password"
                    name='confirmPassword'
                    labelText='Confirm Password'
                    value={confirmPassword}
                    required
                    onChange={(e)=>setConfirmPassword(e.target.value)}/>
                <Button className="my-2" type='submit' variant="primary">
                    Register
                </Button>    
            </Form>
            <Row className="py-3">
                <Col>
                    Have an Account? <Link to={redirect ? `/login?redirect=${redirect}`:'/login'}>Login</Link>
                </Col>    
            </Row>
        </FormContainer>
        </Col>
        <Col lg={6}>
            <FormContainer>
                <PasswordGenerator/>
            </FormContainer>
            
        </Col>
        </Row>
    </div>
  );
};

export default Register;