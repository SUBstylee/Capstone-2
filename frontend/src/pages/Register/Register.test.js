import { render } from '@testing-library/react';
import Register from './Register.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders ProductDetails component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Register /></BrowserRouter></Provider>
    );

    expect(getByText("Sign Up")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Register /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});