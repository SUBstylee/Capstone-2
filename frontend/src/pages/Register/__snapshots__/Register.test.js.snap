// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-6"
      >
        <div
          class="FormContainer"
        >
          <div
            class="container"
          >
            <div
              class="justify-content-md-center row"
            >
              <div
                class="col-md-6 col-12"
              >
                <h1>
                  Sign Up
                </h1>
                <form
                  class=""
                >
                  <div
                    class="CustomInput"
                  >
                    <input
                      autocomplete="current-name"
                      id="name"
                      name="name"
                      required=""
                      type="text"
                      value=""
                    />
                    <label
                      for="name"
                    >
                      <span
                        style="transition-delay: 0ms;"
                      >
                        F
                      </span>
                      <span
                        style="transition-delay: 50ms;"
                      >
                        u
                      </span>
                      <span
                        style="transition-delay: 100ms;"
                      >
                        l
                      </span>
                      <span
                        style="transition-delay: 150ms;"
                      >
                        l
                      </span>
                      <span
                        style="transition-delay: 200ms;"
                      >
                         
                      </span>
                      <span
                        style="transition-delay: 250ms;"
                      >
                        N
                      </span>
                      <span
                        style="transition-delay: 300ms;"
                      >
                        a
                      </span>
                      <span
                        style="transition-delay: 350ms;"
                      >
                        m
                      </span>
                      <span
                        style="transition-delay: 400ms;"
                      >
                        e
                      </span>
                    </label>
                  </div>
                  <div
                    class="CustomInput"
                  >
                    <input
                      autocomplete="current-email"
                      id="email"
                      name="email"
                      required=""
                      type="email"
                      value=""
                    />
                    <label
                      for="email"
                    >
                      <span
                        style="transition-delay: 0ms;"
                      >
                        E
                      </span>
                      <span
                        style="transition-delay: 50ms;"
                      >
                        m
                      </span>
                      <span
                        style="transition-delay: 100ms;"
                      >
                        a
                      </span>
                      <span
                        style="transition-delay: 150ms;"
                      >
                        i
                      </span>
                      <span
                        style="transition-delay: 200ms;"
                      >
                        l
                      </span>
                      <span
                        style="transition-delay: 250ms;"
                      >
                         
                      </span>
                      <span
                        style="transition-delay: 300ms;"
                      >
                        A
                      </span>
                      <span
                        style="transition-delay: 350ms;"
                      >
                        d
                      </span>
                      <span
                        style="transition-delay: 400ms;"
                      >
                        d
                      </span>
                      <span
                        style="transition-delay: 450ms;"
                      >
                        r
                      </span>
                      <span
                        style="transition-delay: 500ms;"
                      >
                        e
                      </span>
                      <span
                        style="transition-delay: 550ms;"
                      >
                        s
                      </span>
                      <span
                        style="transition-delay: 600ms;"
                      >
                        s
                      </span>
                    </label>
                  </div>
                  <div
                    class="CustomInput"
                  >
                    <input
                      autocomplete="new-password"
                      id="password"
                      name="password"
                      required=""
                      type="password"
                      value=""
                    />
                    <label
                      for="password"
                    >
                      <span
                        style="transition-delay: 0ms;"
                      >
                        P
                      </span>
                      <span
                        style="transition-delay: 50ms;"
                      >
                        a
                      </span>
                      <span
                        style="transition-delay: 100ms;"
                      >
                        s
                      </span>
                      <span
                        style="transition-delay: 150ms;"
                      >
                        s
                      </span>
                      <span
                        style="transition-delay: 200ms;"
                      >
                        w
                      </span>
                      <span
                        style="transition-delay: 250ms;"
                      >
                        o
                      </span>
                      <span
                        style="transition-delay: 300ms;"
                      >
                        r
                      </span>
                      <span
                        style="transition-delay: 350ms;"
                      >
                        d
                      </span>
                    </label>
                  </div>
                  <div
                    class="CustomInput"
                  >
                    <input
                      autocomplete="new-password"
                      id="confirmPassword"
                      name="confirmPassword"
                      required=""
                      type="password"
                      value=""
                    />
                    <label
                      for="confirmPassword"
                    >
                      <span
                        style="transition-delay: 0ms;"
                      >
                        C
                      </span>
                      <span
                        style="transition-delay: 50ms;"
                      >
                        o
                      </span>
                      <span
                        style="transition-delay: 100ms;"
                      >
                        n
                      </span>
                      <span
                        style="transition-delay: 150ms;"
                      >
                        f
                      </span>
                      <span
                        style="transition-delay: 200ms;"
                      >
                        i
                      </span>
                      <span
                        style="transition-delay: 250ms;"
                      >
                        r
                      </span>
                      <span
                        style="transition-delay: 300ms;"
                      >
                        m
                      </span>
                      <span
                        style="transition-delay: 350ms;"
                      >
                         
                      </span>
                      <span
                        style="transition-delay: 400ms;"
                      >
                        P
                      </span>
                      <span
                        style="transition-delay: 450ms;"
                      >
                        a
                      </span>
                      <span
                        style="transition-delay: 500ms;"
                      >
                        s
                      </span>
                      <span
                        style="transition-delay: 550ms;"
                      >
                        s
                      </span>
                      <span
                        style="transition-delay: 600ms;"
                      >
                        w
                      </span>
                      <span
                        style="transition-delay: 650ms;"
                      >
                        o
                      </span>
                      <span
                        style="transition-delay: 700ms;"
                      >
                        r
                      </span>
                      <span
                        style="transition-delay: 750ms;"
                      >
                        d
                      </span>
                    </label>
                  </div>
                  <button
                    class="my-2 btn btn-primary"
                    type="submit"
                  >
                    Register
                  </button>
                </form>
                <div
                  class="py-3 row"
                >
                  <div
                    class="col"
                  >
                    Have an Account? 
                    <a
                      href="/login?redirect=/"
                    >
                      Login
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-lg-6"
      >
        <div
          class="FormContainer"
        >
          <div
            class="container"
          >
            <div
              class="justify-content-md-center row"
            >
              <div
                class="col-md-6 col-12"
              >
                <div
                  class="PasswordGenerator"
                >
                  <h2>
                    Password Generator
                  </h2>
                  <div
                    class="resultContainer"
                  >
                    <span
                      id="result"
                    />
                    <button
                      class="btn"
                      id="clipboard"
                    >
                      <i
                        class="far fa-clipboard"
                      />
                    </button>
                  </div>
                  <div
                    class="settings"
                  >
                    <div
                      class="setting"
                    >
                      <label
                        for="length"
                      >
                        Password Length (6-20)
                      </label>
                      <input
                        id="length"
                        max="20"
                        min="6"
                        type="number"
                        value="10"
                      />
                    </div>
                    <div
                      class="setting"
                    >
                      <label
                        for="uppercase"
                      >
                        Include uppercase letters
                      </label>
                      <input
                        checked=""
                        id="uppercase"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="setting"
                    >
                      <label
                        for="lowercase"
                      >
                        Include lowercase letters
                      </label>
                      <input
                        checked=""
                        id="lowercase"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="setting"
                    >
                      <label
                        for="numbers"
                      >
                        Include numbers
                      </label>
                      <input
                        checked=""
                        id="numbers"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="setting"
                    >
                      <label
                        for="symbols"
                      >
                        Include symbols
                      </label>
                      <input
                        checked=""
                        id="symbols"
                        type="checkbox"
                      />
                    </div>
                  </div>
                  <button
                    class="btn btn-large"
                    id="generate"
                  >
                    Generate Password
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
